Fix two virtual button bugs

colinator27 opened the pull request:
- Background virtual button sometimes would be sorted incorrectly due to its priority being `-infinity`. (Seems like a GameMaker problem?): this PR forces it to always be at the end of the array
- input_virtual_destroy_all() didn't work at all, used the wrong variable name

Note that the first fix is incorrect if the background virtual button is supposed to be able to have its priority changed. If that's the case, feel free to remove those changes and simply change the `-infinity` on the background button to an arbitrarily-small number like `-1000000`.

colinator27 committed:
Fix two virtual button bugs

- Background virtual button sometimes would be sorted incorrectly due to its priority. (Seems like a GameMaker problem?)
- input_virtual_destroy_all() didn't work at all, used the wrong variable name

JujuAdams commented:
What version of GameMaker did you test on for the first bug (relating to issues with `array_sort()`)?

colinator27 committed:
Merge branch 'JujuAdams:dev_6.0' into dev_6.0

colinator27 committed:
Cleaner fix for array_sort issue

This gets rid of the previous fix, and instead just adds sign() to the sorting function

colinator27 commented:
Updated with a cleaner fix for the `array_sort()` problem. Seems like it may just be a very obscure misuse of the function rather than a GM bug, since it's supposed to only handle integers (and `infinity` isn't quite an integer). So, all it takes is adding `sign()` to the sorting function and it works fine.

In case anyone cares about this in the future, this `array_sort()` issue can be reproduced with this small snippet (at least on version 2023.6):
```gml
var array = [100, 100, 100, -infinity];
array_sort(array, function(a, b) { return b - a; });
show_debug_message(array); // outputs [ 100,100,-inf,100 ] rather than [ 100,100,100,-inf ]
```
This, too, can be easily fixed by adding `sign()` to the sorting function.
